server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -1106,4 +1106,8 @@ public static boolean assertCurrentMethodIsNotCalledRecursively() {
         }
         return true;
     }
+
```

Comment:
I don't know what's the best place for this. I've added it here. It could be also in a separate file in the same package, or entirely some where else. Suggestions are welcomed.